File: core/evidence/semantic_orthodoxy_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def detect_semantic_orthodoxy(interpretations: List[Dict[str, Any]], depth: int) -> Dict[str, Any]:
    orthodox = len(interpretations) <= 1 and depth >= 3
    return {"orthodoxy_detected": orthodox, "suppress": orthodox, "orthodoxy_pressure": {"level": 0.85 if orthodox else 0.0}}

File: src/evidence/semanticOrthodoxyEngine.ts
/**
 * Converted from Python: core/evidence/semantic_orthodoxy_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function detectSemanticOrthodoxy(interpretations: any, depth: any): any {
  var orthodox: any = py.and2((py.len(interpretations) <= 1), () => ((depth >= 3)));
  return {"orthodoxy_detected": orthodox, "suppress": orthodox, "orthodoxy_pressure": {"level": (py.truthy(orthodox) ? py.F(0.85) : py.F(0.0))}};
}
